src/app/modules/main/communities/models/discord_categories_model.nim

```diff
@@ -107,3 +107,10 @@ QtObject:
       let index = self.createIndex(idx, 0, nil)
       self.items[idx].selected = true
       self.dataChanged(index, index, @[ModelRole.Selected.int])
+
+  proc selectOneItem*(self: DiscordCategoriesModel, id: string) =
```

Comment:
This is impossible in NIM imo :) Also, here we have a unique selection, one item at most (radio buttons)

Reply:
Imo it is, because than it lives exclusively on UI side. It also gives full independence between the actual model and the selection that is done on top. The qml type is https://doc.qt.io/qt-5/qml-qtqml-models-itemselectionmodel.html, it nicely manages selection, including observing the underlying object and reacting on removals, moves and so on. Additional profit is that UI side is really on ui side and when used in Storybook, you don't have to reimplement selection to have the component functional.

Reply:
Ok then, if you insist - please file a separate issue for the backend part of the corresponding epic; I merely extended the existing NIM code to make the UI functional

Reply:
No, I'm not insisting, just pointing that it's not impossible. Moving it to a separate task is definitely a good idea.

Reply:
here you go: https://github.com/status-im/status-desktop/issues/12099 :)